packages/lesswrong/components/ultraFeed/UltraFeed.tsx

```diff
@@ -0,0 +1,228 @@
+        <span className={classes.titleTextMobile}>The Feed</span>
+      </span>
+    </div>
+    <div className={classes.settingsButtonContainer}>
+      <SettingsButton 
+        showIcon={true}
+        onClick={toggleSettings}
+      />
+    </div>
+  </>;
+
+  return (
+    <div className={classes.root}>
+      <div className={classes.toggleContainer}>
+        <SectionFooterCheckbox 
+          value={ultraFeedEnabled} 
+          onClick={toggleUltraFeed} 
+          label="Use UltraFeed"
+          tooltip="Hide Quick Takes and Popular Comments sections and show a feed of posts and comments from users you subscribe to"
+        />
+      </div>
+      
+      {ultraFeedEnabled && <>
+        <UltraFeedObserverProvider>
+          <SingleColumnSection>
+            {/* place this higher than top feed so it properly scrolls into view */}
+            <SectionTitle title={customTitle} titleClassName={classes.sectionTitle} />
+            {settingsVisible && (
+              <div className={classes.settingsContainer}>
+                <UltraFeedSettings onClose={() => setSettingsVisible(false)} />
+              </div>
+            )}
+            
+            <div className={classes.ultraFeedNewContentContainer}>
+              <MixedTypeFeed
+                resolverName="UltraFeed"
+                sortKeyType="Date"
+                firstPageSize={15}
+                pageSize={15}
+                refetchRef={refetchSubscriptionContentRef}
+                resolverArgsValues={{ sessionId }}
+                loadMoreDistanceProp={1000}
+                renderers={{
+                    feedCommentThread: {
+                      fragmentName: 'FeedCommentThreadFragment',
+                      render: (item: FeedCommentThreadFragment) => {
+                        if (!item) {
+                          return null;
+                        }
+                        
+                        return (
+                          <FeedItemWrapper>
+                            <UltraFeedThreadItem thread={item} />
+                          </FeedItemWrapper>
+                        );
+                      }
+                    },
+                    feedPost: {
+                      fragmentName: 'FeedPostFragment',
+                      render: (item: FeedPostFragment) => {
+                        if (!item) {
+                          return null;
+                        }
+                        
+                        return (
+                          <FeedItemWrapper>
+                            <UltraFeedPostItem post={item.post} postMetaInfo={item.postMetaInfo} />
+                          </FeedItemWrapper>
+                        );
+                      }
+                    },
+                    feedSpotlight: {
+                      fragmentName: 'FeedSpotlightFragment',
+                      render: (item: {_id: string, spotlight: FeedSpotlightFragment['spotlight']}) => {
```

Comment:
any reason the item type here isn't just `FeedSpotlightFragment` (which contains both an `_id` and `spotlight` field)?  Can then just trivially destructure it inside the function without needing to manually annotate the types of the individual fields:
```
const { _id, spotlight } = item;
```